hibernate-core/src/main/java/org/hibernate/dialect/MySQLDialect.java

```diff
@@ -635,7 +635,7 @@ public void initializeFunctionRegistry(FunctionContributions functionContributio
 		SqmFunctionRegistry functionRegistry = functionContributions.getFunctionRegistry();
 
 		functionRegistry.noArgsBuilder( "localtime" )
-				.setInvariantType(basicTypeRegistry.resolve( StandardBasicTypes.TIMESTAMP ))
+				.setInvariantType(basicTypeRegistry.resolve( StandardBasicTypes.TIME ))
 				.setUseParenthesesWhenNoArgs( false )
 				.register();
```

Comment:
But you should be able to simply _remove_ this stuff, no, since we already have:

```java
		functionRegistry.noArgsBuilder( "localtime" )
				.setInvariantType(timeType)
				.setUseParenthesesWhenNoArgs( false )
				.register();
```

being called in `CommonFunctionFactory.localtimeLocaltimestamp()`